core/arch/x86/x86.asm

```diff
@@ -604,6 +604,9 @@ cat_done_saving_dstack:
         CALLC0(GLOBAL_REF(get_cleanup_and_terminate_global_do_syscall_entry))
 #if defined(MACOS) && !defined(X64)
         lea      REG_XSP, [-2*ARG_SZ + REG_XSP] /* maintain align-16 w/ 2 pushes below */
+#endif
+#if defined(LINUX) && !defined(X64)
+        lea      REG_XSP, [-ARG_SZ + REG_XSP] /* maintain align-16 w/ 2 pushes below */
```

Comment:
If you could help me understand: why is Linux different from Mac?  At a glance aren't they the same below with 2 pushes?

Reply:
It seems to me that stack is misaligned for previous call too. For instance, `get_cleanup_and_terminate_global_do_syscall_entry`. I need further investigation.

Reply:
I will continue on this pull request after some sleep.

Reply:
Sure, np